File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlDistinct.cs
namespace T1.SqlSharp.Expressions;

public class SqlDistinct : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.Distinct;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_Distinct(this);
    }

    public required ISqlExpression Value { get; set; }

    public string ToSql()
    {
        return $"DISTINCT {Value.ToSql()}";
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlDistinct.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlDistinct implements ISqlExpression {
    SqlType: SqlType = SqlType.Distinct;
    Span: TextSpan = new TextSpan();
    Value!: ISqlExpression;

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_Distinct(this);
    }

    ToSql(): string {
        return `DISTINCT ${this.Value.ToSql()}`;
    }
} 